src/main/java/com/liefery/android/stop_badge/StopBadgeFactory.java

```diff
@@ -0,0 +1,38 @@
+package com.liefery.android.stop_badge;
+
+import android.annotation.SuppressLint;
+import android.graphics.*;
+import android.util.Log;
+import android.util.LruCache;
+
+@SuppressLint( "NewApi" )
+public class StopBadgeFactory {
+
+    private static StopBadgeFactory instance = null;
+
+    private BadgeCache cache = new BadgeCache();
+
+    protected StopBadgeFactory() {
+        // Exists only to defeat instantiation.
+    }
+
+    public static Bitmap getBitmap( StopBadge badge, int size ) {
+        if ( instance == null ) {
+            instance = new StopBadgeFactory();
+        }
+
+        if ( instance.cache.get( badge.toKey( size ) ) == null ) {
+            Bitmap newBadge = badge.export( size );
+            instance.cache.put( badge.toKey( size ), newBadge );
+            return newBadge;
+        } else {
+            return instance.cache.get( badge.toKey( size ) );
+        }
+    }
+
+    public class BadgeCache extends LruCache<String, Bitmap> {
+        BadgeCache() {
+            super( 2 * 1024 * 1024 ); // 2MiB
```

Comment:
There could potentially be up to 200 badges shown on a map. With 20dp in size + shadow. That would add up to 10mb of Bitmap data 😱 But I guess the cache size will be our least problem then. Rendering and caching on the main thread will slow down our UI dramatically.

Long story short: could we use a low percentage of the device's memory (rather than a fixed value)? Or make it devices density level dependent?